src/CommandingOfficers/Tech.java

```diff
@@ -0,0 +1,514 @@
+
+    @Override
+    protected void perform(MapMaster gameMap)
+    {
+      // Bump up our power level.
+      myCommander.addCOModifier(damageBuff);
+      myCommander.addCOModifier(defenseBuff);
+
+      // Overcharge
+      for(Unit u: myCommander.units)
+      {
+        // Only mechanical units get overcharged.
+        if( u.model.isNone(UnitModel.TROOP) )
+        u.alterHP(healAmount, true);
+      }
+    }
+  }
+
+  /** Drop a MechWarrior onto the most contested point on the battlefront. */
+  private static class TechdropAbility extends CommanderAbility
+  {
+    private static final long serialVersionUID = 1L;
+
+    private CODamageModifier damageBuff = null;
+    private CODefenseModifier defenseBuff = null;
+    private int dropRange;
+    private int numDrops;
+    private UnitModel unitModelToDrop;
+
+    GameEventQueue abilityEvents;
+
+    TechdropAbility(Commander commander, String abilityName, double abilityCost, UnitModel unitToDrop, int buff, int num, int abilityRange)
+    {
+      super(commander, abilityName, abilityCost);
+
+      unitModelToDrop = unitToDrop;
+
+      // Create COModifiers that we can apply when needed.
+      damageBuff = new CODamageModifier(buff);
+      defenseBuff = new CODefenseModifier(buff);
+      numDrops = num;
+      dropRange = abilityRange;
+      abilityEvents = new GameEventQueue();
+
+      // Only mechanical units get the firepower boost.
+      long exclude = UnitModel.TROOP;
+      ArrayList<UnitModel> models = commander.getAllModels(Tech.mechanicalUnits, true, exclude);
+      for( UnitModel m : models )
+        damageBuff.addApplicableUnitModel(m);
+    }
+
+    @Override
+    protected void perform(MapMaster gameMap)
+    {
+      // Bump up our power level.
+      myCommander.addCOModifier(damageBuff);
+      myCommander.addCOModifier(defenseBuff);
+
+      // Perform any events we generated in getEvents().
+      while(!abilityEvents.isEmpty())
+      {
+        abilityEvents.poll().performEvent(gameMap);
+      }
+    }
+
+    @Override
+    public GameEventQueue getEvents(MapMaster gameMap)
+    {
+      boolean log = true;
+      if( log ) System.out.println("[TechDrop] getEvents() entry");
+
+      // Prep events for `numDrops` deployments. Recalculate landing position between each.
+      Set<XYCoord> dropLocs = new HashSet<XYCoord>();
+      for( int i = 0; i < numDrops; ++i )
+      {
+        TeleportEvent techDrop = generateDropEvent(gameMap, dropLocs, log);
+        if( null != techDrop )
+        {
+          // Store the new event.
+          dropLocs.add(techDrop.getEndPoint());
+          abilityEvents.add(techDrop);
+        }
+        else
+        {
+          System.out.println("[TechdropAbility.getEvents] WARNING! Could not find drop location.");
+        }
+      }
+
+      return abilityEvents;
+    }
+
+    private TeleportEvent generateDropEvent(MapMaster gameMap, Set<XYCoord> priorDrops, boolean log)
```

Comment:
I think this logic could be simplified a lot using the `MassStrikeUtils` thing I set up in the test branch.
It probably would be a bit less efficient, but I think the gains in readability would be significant.
I can make a PR on this PR for that, if you'd like? (Probably wouldn't make it until I'm done reviewing, though.)

Reply:
I'm hesitant to agree to that given that a) I'm unfamiliar with how `MassStrikeUtils` works, and b) this already does.
It would be worth bringing that over for the future though, I'm sure. Had it been part of `master` already, I would have at least tried to use it, if it is flexible enough.

Reply:
b) it does... what?
Mostly, it would focus the valuation function on scoring a *single* space, instead of being integrated with a bunch of loops evaluating many spaces.

Reply:
b) ... works.
Next time I build a globally-targetable ability, maybe I'll try to use the utils, though this PR may be evidence that I like to tweak/fine-tune/customize things beyond what is often easily generalizeable.

Reply:
I'd like to take a stab at porting it.
Do you mind?